sdk/src/main/java/io/opentelemetry/sdk/trace/TracerSdkProvider.java

```diff
@@ -138,7 +138,7 @@ public void forceFlush() {
 
     private Clock clock = MillisClock.getInstance();
     private IdsGenerator idsGenerator = new RandomIdsGenerator();
-    private Resource resource = EnvVarResource.getResource();
+    private Resource resource = Resource.getTelemetrySdk().merge(EnvVarResource.getResource());
```

Comment:
So when a user wants to define their own `Resource`, they always have to merge the default one to get these values. I think this is OK, but it could also be good independently of this PR to think if `Resource` would make sense to have its own SPI, so that any `ResourceProvider` in the app are automatically registered without having to manually merge them all. Of course, there could be ordering problems with this, but it's probably still simpler than manual merging in many cases.

Reply:
Let's definitely address this case in another PR (we have meant to implement the `Resource` part you mention, in some way or another, for some time now, so hopefully we will have it soon).